include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,406 @@
+// ==========================================================================
+
+// NOTE(rrahn): In-file forward for function call operator.
+struct VersionCheck;
+inline bool _checkForNewerVersion(VersionCheck &);
+inline std::string _getOS();
+inline std::string _getPath();
+inline std::string _getBitSys();
+
+// ==========================================================================
+// Tags, Classes, Enums
+// ==========================================================================
+
+struct VersionControlTags
+{
+    constexpr static const char * SEQAN_NAME = "seqan";
+    constexpr static const char * UNREGISTERED_APP = "UNREGISTERED APP";
+};
+
+enum VersionControlOptions
+{
+    VERSION_OPTION_DEV      = 0,
+    VERSION_OPTION_APP_ONLY = 1,
+    VERSION_OPTION_OFF      = 2
+};
+
+struct VersionCheck
+{
+    // ----------------------------------------------------------------------------
+    // Member Variables
+    // ----------------------------------------------------------------------------
+    std::string _url;
+    std::string _name;
+    std::string _version = "0.0.0";
+    std::string _program;
+    std::string _command;
+    std::string _website = "https://github.com/seqan/seqan/tree/master/apps";
+    std::string _path = _getPath();
+
+    // ----------------------------------------------------------------------------
+    // Constructors
+    // ----------------------------------------------------------------------------
+    VersionCheck(std::string const & name,
+                 std::string const & version,
+                 std::string const & website)
+    {
+        _name = name;
+        if (!version.empty() &&
+            std::regex_match(version, std::regex("^[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+.*")))
+            _version = version.substr(0,5); // in case the git revision number is given take only version number
+        if (!website.empty())
+            _website = website;
+        _url = "http://www.seqan.de/version_check/SeqAn_" + _getOS() + _getBitSys() + _name + "_" + _version;
+        _getProgram();
+        _updateCommand();
+    }
+
+    // ----------------------------------------------------------------------------
+    // Member Functions
+    // ----------------------------------------------------------------------------
+#if defined(PLATFORM_WINDOWS)
+    void _getProgram()
+    {
+        _program = "powershell.exe -NoLogo -NonInteractive -Command \"& {Invoke-WebRequest -erroraction 'silentlycontinue' -OutFile";
+    }
+#else  // Unix based platforms.
+    void _getProgram()
+    {
+        // ask if system call for version or help is successfull
+        if (!system("wget --version > /dev/null 2>&1"))
+            _program = "wget -q -O";
+        else if (!system("curl --version > /dev/null 2>&1"))
+            _program =  "curl -o";
+#ifndef __linux  // ftp call does not work on linux
+        else if (!system("which ftp > /dev/null 2>&1"))
+            _program =  "ftp -Vo";
```

Comment:
Do we have a plan to provide multiple servers(http and ftp ..) for call-home functionality?
Just curious.  :)